libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -27,14 +27,17 @@ internal class MapboxRerouteController(
     private val routeOptionsUpdater: RouteOptionsUpdater,
     private val rerouteOptions: RerouteOptions,
     threadController: ThreadController,
-) : NavigationRerouteController {
+) : NavigationRerouteController, RerouteOptionsAdapterManager {
 
     private val observers = CopyOnWriteArraySet<RerouteController.RerouteStateObserver>()
 
     private val mainJobController: JobControl = threadController.getMainScopeAndRootJob()
 
     private var requestId: Long? = null
 
+    @Volatile
```

Comment:
There's shouldn't be a need for this, as far as I can tell all of the interaction with this object is expected on the main thread.